test/Transforms/IndVarSimplify/preserve-phi-when-replacement-is-in-loop.ll

```diff
@@ -0,0 +1,46 @@
+; RUN: opt < %s -indvars -S | FileCheck %s
+
+; The inner loop (%header1) has a fixed trip count.
+; The indvars pass is tempted to delete the phi instruction %hexit,
+; and replace its uses with %add3.
+; But %hexit is used in %latch0, which is outside the inner loop and
+; its exit block. Deleting the phi %hexit would break LCSSA form.
+
+; CHECK: @main
+; CHECK: exit1:
+; CHECK-NEXT: %hexit = phi i32 [ %hnext, %header1 ]
+
```

Comment:
Should we also add a check for the `latch0` label (or a regex to match any label) to ensure that the `phi` occurs before the end of the block?

Reply:
The `CHECK-NEXT` makes sure the match is on the next line after the `exit1:`; in other words it makes sure the phi is the first instruction in the exit1 block.

Reply:
I'll add more check lines to make it clearer though.

Reply:
Ha! No, that's right, I just missed that it was check-next. My poor reading skills.

Reply:
A failure looks like this:

```
FAIL: LLVM :: HLSL/passes/indvars/preserve-phi-when-replacement-is-in-loop.ll (2323 of 3935)
******************** TEST 'LLVM :: HLSL/passes/indvars/preserve-phi-when-replacement-is-in-loop.ll' FAILED ********************
Script:
--
/project/dxc/build.release/./bin/opt < /project/DirectXShaderCompiler/test/HLSL/passes/indvars/preserve-phi-when-replacement-is-in-loop.ll -indvars -S | /project/dxc/build.release/./bin/FileCheck /project/DirectXShaderCompiler/test/HLSL/passes/indvars/preserve-phi-when-replacement-is-in-loop.ll
--
Exit Code: 1

Command Output (stderr):
--
/project/DirectXShaderCompiler/test/HLSL/passes/indvars/preserve-phi-when-replacement-is-in-loop.ll:11:15: error: expected string not found in input
; CHECK-NEXT: %hexit = phi i32 [ %hnext, %header1 ]
              ^
<stdin>:26:8: note: scanning from here
exit1: ; preds = %header1
       ^
<stdin>:30:2: note: possible intended match here
 %useh = phi i32 [ %h0, %header0 ], [ %add3, %exit1 ]
 ^
```